### DISCUSSION TITLE
How does the mod compatibility check work? I want to implement it on linux

### DISCUSSION THREAD

**Discussion by tharbad08** [X2CommunityCore/xcom2-launcher#284] [General]
**Description**
I know there is another linux thread already.
I don't know dot net but I do know bash and python. So I'm thinking about scripting the mod compatibility check.
Can someone explain how it works?

My script won't be a new launcher but it'll help debugging and maybe even generate XComModOptions.ini.

**Suggest a solution**
That remain to be seen.

**Comment by RevZero** [comment]
I will move this to discussions section as this is not directly related to AML.

**Comment by RevZero** [comment]
The type of conflicts AML is able to detect are duplicate mod class overrides, also know as MCO-conflicts.
More information here: https://www.reddit.com/r/xcom2mods/wiki/index/mod_class_overrides/

**Comment by tharbad08** [comment]
@RevZero Can you point me to the source files where those checks are implemented? I prefer to adapt that algorithm than creating a new one.

**Reply by RevZero** [reply]
Step one would be to get the actual override for each mod
https://github.com/X2CommunityCore/xcom2-launcher/blob/d23f80acfeb7f556c79d8853dda8ed311b8a43f0/xcom2-launcher/xcom2-launcher/Classes/Mod/ModEntry.cs#L202

Then determine the conflicts
https://github.com/X2CommunityCore/xcom2-launcher/blob/d23f80acfeb7f556c79d8853dda8ed311b8a43f0/xcom2-launcher/xcom2-launcher/Classes/Mod/ModList.cs#L76

Maybe you can work with that.
